SignalMessaging/contacts/OWSContactsManager.m

```diff
@@ -566,8 +565,12 @@ - (NSAttributedString *)formattedFullNameForRecipientId:(NSString *)recipientId
         NSAttributedString *lastName =
             [[NSAttributedString alloc] initWithString:cachedLastName attributes:lastNameAttributes];
 
+        CNMutableContact *formatContact = [CNMutableContact new];
+        formatContact.givenName = firstName.string;
+        formatContact.familyName = lastName.string;
+        CNContactDisplayNameOrder nameOrder = [CNContactFormatter nameOrderForContact:formatContact];
```

Comment:
Docs are opaque about why `[CNContactFormatter nameOrderForContact:]` takes a contact. https://developer.apple.com/documentation/contacts/cncontactformatter/1397731-nameorderforcontact

Reply:
> Can you verify this works with the various combinations of display vs. sort in Settings > Contacts?

Yup, this works.

> OWSContact *contact = [self contactForRecipientId:recipientId];

We use this method - `formattedFullNameForRecipientId` - to format recipient ids, for which we don't necessarily have a system contact.

> And in general, when we're touching CNContact*anything, I'd try to put it in our contact adapter. 

Ditto - there isn't necessarily an instance of `Contact`.

Reply:
I've modified this to _try_ to use the `CNContact` if we have one, but to fall back to constructing one.